### ISSUE TITLE
podman generate kube does not generate `env:` variable

### ISSUE THREAD

**Issue by Taymindis** [containers/podman#12781]
/kind bug

Description

## Steps to reproduce the issue:
```
podman run --name simple-nginx -e ABCUSER="xxx" --pod service-lab -d nginx
podman generate kube service-lab

```

### output

```yml
 Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-3.4.4
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2022-01-09T15:48:20Z"
  labels:
    app: service-lab
  name: service-lab
spec:
  containers:
  - args:
    - nginx
    - -g
    - daemon off;
    image: docker.io/library/nginx:latest
    name: simple-nginx
    resources: {}
    securityContext:
      capabilities:
        drop:
        - CAP_MKNOD
        - CAP_NET_RAW
        - CAP_AUDIT_WRITE
  restartPolicy: Never
status: {}

```

env `ABCUSER` variable not found

```
podman --version
podman version 3.4.4

```

**Comment by rhatdan** [comment]
@umohnani8 PTAL

**Comment by rhatdan** [comment]
@Taymindis Interested in opening a PR to fix?

**Comment by Taymindis** [comment]
> @Taymindis Interested in opening a PR to fix?

@rhatdan 

Thanks for inviting but I am afraid that I'm not capable to fix it.

**Comment by dannykansas** [comment]
@Taymindis is there a corollary feature in Docker (or elsewhere) that you could point to for expectations around this feature?

**Comment by rhatdan** [comment]
Docker does not have a generate kube feature.

**Comment by rhatdan** [comment]
I just ran a test for this with the main branch and it looks like it works correctly to me. Seems to be fixed
```
$ ./bin/podman pod create --name service-lab
e5238013a7037420ae519ceda7c71f507f764aa17654f421821c80f38272bea3
$ ./bin/podman create --name test -e ABCUSER="xxx" --pod service-lab alpine
a873e749750ec8c7728f2616853ad6e1ba558be7912075c434269888767ea653
$ ./bin/podman generate kube service-lab
# Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-4.0.0-dev
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2022-01-10T19:38:08Z"
  labels:
    app: service-lab
  name: service-lab
spec:
  containers:
  - env:
    - name: ABCUSER
      value: xxx
    image: docker.io/library/alpine:latest
    name: test
    resources: {}
    securityContext:
      capabilities:
        drop:
        - CAP_MKNOD
        - CAP_NET_RAW
        - CAP_AUDIT_WRITE
  restartPolicy: Never
status: {}
```
Should be fixed in podman 4.0, which Release candidates are being generated this week.
